uh50_api/uh50_reader.py

```diff
@@ -0,0 +1,69 @@
+""" 
+Reads raw UH50 response data from the UH50 unit. 
+To test the connection use validate, which will return the model name.
+"""
+import serial
+from serial import Serial
+
+
+class UH50Reader:
+    def __init__(self, port) -> None:
+        self._port = port
+        self.model: str
+
+    def validate(self) -> str:
+        "Open connection to the UH50 and get the model name, thereby validating the connection"
+        with self._connect_serial() as conn:
+            self._wake_up(conn)
+        return self.model

```

Comment:
Dit is lastige code om te lezen. Om te bepalen wat er in self.model zit, moet je de _wake_up functie bekijken. Die functie heeft als side effect het zetten van dat field. Wanneer je het modelnumber zou returnen, hoef je die hele functie niet in om te begrijpen wat hier gebeurt. 

Daarnaast blijft ik het nog wel vreemd vinden dat die wakeup call het model number teruggeeft. Is dat een quirk van die meter? Moet je die eerst wakker maken voor je er verder iets mee kunt doen en spuugt ie dan gelijk het model number uit ofzo? Of zijn die dingen ook nog los te trekken van elkaar?